readme.md

```diff
@@ -54,7 +54,15 @@ With the activation of the [Arithmetic Operation Cost Limit](https://github.com/
 
 Alternatively, this proposal could raise the limit to a higher constant value like `258`, the constant selected by Satoshi Nakamoto in [`reverted makefile.unix wx-config -- version 0.3.6`](https://gitlab.com/bitcoin-cash-node/bitcoin-cash-node/-/commit/757f0769d8360ea043f469f3a35f6ec204740446) (July 29, 2010). However, because the limit is no longer necessary for mitigating worst-case transaction validation cost, selection of any particular constant would be arbitrary.
 
-By fully-removing the limit, overall protocol complexity is reduced, simplifying both future VM implementations and contract development. For VM implementations, eliminating out-of-range cases significantly reduces the combinatorial set of possible inputs and outputs for the numeric operations: `OP_1ADD` (`0x8b`), `OP_1SUB` (`0x8c`), `OP_NEGATE` (`0x8f`), `OP_ABS` (`0x90`), `OP_NOT` (`0x91`), `OP_0NOTEQUAL` (`0x92`), `OP_ADD` (`0x93`), `OP_SUB` (`0x94`), `OP_MUL` (`0x95`), `OP_DIV` (`0x96`), `OP_MOD` (`0x97`), `OP_BOOLAND` (`0x9a`), `OP_BOOLOR` (`0x9b`), `OP_NUMEQUAL` (`0x9c`), `OP_NUMEQUALVERIFY` (`0x9d`), `OP_NUMNOTEQUAL` (`0x9e`), `OP_LESSTHAN` (`0x9f`), `OP_GREATERTHAN` (`0xa0`), `OP_LESSTHANOREQUAL` (`0xa1`), `OP_GREATERTHANOREQUAL` (`0xa2`), `OP_MIN` (`0xa3`), `OP_MAX` (`0xa4`), and `OP_WITHIN` (`0xa5`). For contract authors, eliminating the possibility of out-of-range errors prevents a class of potential vulnerabilities arising from a contract system's failure to validate that intermediate arithmetic results never exceed an (uncommonly-encountered) maximum number length limit.
+### Interaction with Existing Operations
+
+With one exception regarding overflow, the behavior of every operation remains unchanged. The overflow exception is described in [Removal of Overflow Behavior](#removal-of-overflow-behavior).
+
+The following list is the complete set of operations that interact with data on the stack with the assumption that the input(s) and/or output are VM numbers: `OP_NUM2BIN` (`0x80`), `OP_BIN2NUM` (`0x81`), `OP_1ADD` (`0x8b`), `OP_1SUB` (`0x8c`), `OP_NEGATE` (`0x8f`), `OP_ABS` (`0x90`), `OP_NOT` (`0x91`), `OP_0NOTEQUAL` (`0x92`), `OP_ADD` (`0x93`), `OP_SUB` (`0x94`), `OP_MUL` (`0x95`), `OP_DIV` (`0x96`), `OP_MOD` (`0x97`), `OP_BOOLAND` (`0x9a`), `OP_BOOLOR` (`0x9b`), `OP_NUMEQUAL` (`0x9c`), `OP_NUMEQUALVERIFY` (`0x9d`), `OP_NUMNOTEQUAL` (`0x9e`), `OP_LESSTHAN` (`0x9f`), `OP_GREATERTHAN` (`0xa0`), `OP_LESSTHANOREQUAL` (`0xa1`), `OP_GREATERTHANOREQUAL` (`0xa2`), `OP_MIN` (`0xa3`), `OP_MAX` (`0xa4`), and `OP_WITHIN` (`0xa5`).
```

Comment:
opcodes are assuming input is VM number but they're *guaranteeing* output will be a VM number because they are specified to return a VM number so it's not correct to say that they're assuming the output, suggest:

`The following list is the complete set of operations that interact with data on the stack with the assumption that the input(s) are VM numbers, and any returned value will be a VM number:`

Reply:
be specific about overflow:

>With one exception regarding operands overflow

we still check the output for overflow and fail if bigger than MAX_SIZE (value of magnitude greater than 2^79999-1)

Reply:
The original wording was carefully crafted and I think it's correct. The point is that all of these OPs deal with VM numbers in one way or another, and it doesn't matter if it's input or output. For example I don't think `OP_NUMEQUALVERIFY` guarantees a VM number output, does it?

I will start a separate thread for your other comment about overflow

Reply:
yeah, if there's an output it *will* be a minimally encoded script number

and OP_NUMEQUALVERIFY is the only exception with no return, all others return exactly 1 numerical value

Reply:
But if there is actually overflow then... need to reconsider this content.